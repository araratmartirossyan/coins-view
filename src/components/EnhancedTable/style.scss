.tool-bar {
  flex-direction: column;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-left: 16px;
  padding-right: 8px;

  border-bottom: 1px solid rgb(241, 241, 241);
}

.tool-bar__title {
  align-self: flex-start;
}

// table

.enhanced-table {
  margin: 0 auto;
  margin-top: 40px;
  width: 800px;
}

.enhanced-table__container {
  max-height: 440px;
}

.enhanced-table__paper {
  width: 100%;
  margin-bottom: 16px;
}

.enhanced-table__table {
  min-width: 600px;
}

.visuallyHidden {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  top: 20px;
  width: 1px;
}
