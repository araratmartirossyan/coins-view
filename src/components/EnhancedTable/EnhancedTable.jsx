import React from "react";
import PropTypes from "prop-types";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TablePagination from "@material-ui/core/TablePagination";
import TableRow from "@material-ui/core/TableRow";
import TableSortLabel from "@material-ui/core/TableSortLabel";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Switch from "@material-ui/core/Switch";

import ButtonsControls from "../ButtonsControls/ButtonsControls";
import { getComparator, stableSort } from "../../utils/helpers";

import "./style.scss";

const headCells = [
  { id: "pair", numeric: false, disablePadding: true, label: "Pair" },
  {
    id: "lastPrice",
    numeric: true,
    disablePadding: false,
    label: "Last Price",
  },
  { id: "change", numeric: true, disablePadding: false, label: "Change" },
];

const EnhancedTableHead = ({ order, orderBy, onRequestSort }) => {
  const createSortHandler = (property) => (event) => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        <TableCell padding="checkbox">
          <Checkbox disabled />
        </TableCell>
        {headCells.map(({ id, numeric, disablePadding, label }) => (
          <TableCell
            key={id}
            align={numeric ? "right" : "left"}
            padding={disablePadding ? "none" : "default"}
            sortDirection={orderBy === id ? order : false}
          >
            <TableSortLabel
              active={orderBy === id}
              direction={orderBy === id ? order : "asc"}
              onClick={createSortHandler(id)}
            >
              {label}
              {orderBy === id ? (
                <span className="visuallyHidden">
                  {order === "desc" ? "sorted descending" : "sorted ascending"}
                </span>
              ) : null}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
};

EnhancedTableHead.propTypes = {
  onRequestSort: PropTypes.func.isRequired,
  order: PropTypes.oneOf(["asc", "desc"]).isRequired,
  orderBy: PropTypes.string.isRequired,
};

const EnhancedTableToolbar = ({
  search,
  filter,
  handleChangeSearch,
  toggleShowFavorite,
  changeFilterValue,
  showFavorite,
}) => {
  return (
    <Toolbar className="tool-bar">
      <Typography
        className="tool-bar__title"
        variant="h6"
        id="tableTitle"
        component="div"
      >
        Информация
      </Typography>
      <ButtonsControls
        search={search}
        filter={filter}
        handleChangeSearch={handleChangeSearch}
        changeFilterValue={changeFilterValue}
        toggleShowFavorite={toggleShowFavorite}
        showFavorite={showFavorite}
      />
    </Toolbar>
  );
};

export default function EnhancedTable(props) {
  const {
    handleChangeSearch,
    changeFilterValue,
    handleToggleElement,
    toggleShowFavorite,
    isFavorite,
    search,
    showFavorite,
    priceData,
    filter,
  } = props;

  const [order, setOrder] = React.useState("asc");
  const [orderBy, setOrderBy] = React.useState("calories");
  const [page, setPage] = React.useState(0);
  const [dense, setDense] = React.useState(false);
  const [rowsPerPage, setRowsPerPage] = React.useState(5);

  const rows = priceData;

  const handleRequestSort = (_, property) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  const handleChangeDense = (event) => {
    setDense(event.target.checked);
  };

  const emptyRows =
    rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);

  return (
    <div className="enhanced-table">
      <Paper className="enhanced-table__paper">
        <EnhancedTableToolbar
          search={search}
          filter={filter}
          handleChangeSearch={handleChangeSearch}
          changeFilterValue={changeFilterValue}
          toggleShowFavorite={toggleShowFavorite}
          showFavorite={showFavorite}
        />
        <TableContainer className="enhanced-table__container">
          <Table
            className="enhanced-table__table"
            aria-labelledby="tableTitle"
            size={dense ? "small" : "medium"}
            aria-label="enhanced table"
          >
            <EnhancedTableHead
              order={order}
              orderBy={orderBy}
              onRequestSort={handleRequestSort}
              rowCount={rows.length}
            />
            <TableBody>
              {stableSort(rows, getComparator(order, orderBy))
                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                .map((row, index) => {
                  const labelId = `enhanced-table-checkbox-${index}`;

                  return (
                    <TableRow
                      hover
                      onClick={(event) => handleToggleElement(event, row.id)}
                      role="checkbox"
                      tabIndex={-1}
                      key={row.id}
                    >
                      <TableCell padding="checkbox">
                        <Checkbox
                          checked={isFavorite(row.id)}
                          inputProps={{ "aria-labelledby": labelId }}
                        />
                      </TableCell>
                      <TableCell
                        component="th"
                        id={labelId}
                        scope="row"
                        padding="none"
                      >
                        {row.pair}
                      </TableCell>
                      <TableCell align="right">{row.lastPrice}</TableCell>
                      <TableCell align="right">{`${row.change}%`}</TableCell>
                    </TableRow>
                  );
                })}
              {emptyRows > 0 && (
                <TableRow style={{ height: (dense ? 33 : 53) * emptyRows }}>
                  <TableCell colSpan={6} />
                </TableRow>
              )}
            </TableBody>
          </Table>
        </TableContainer>
        <TablePagination
          rowsPerPageOptions={[5, 10, 25, 50, 100]}
          component="div"
          count={rows.length}
          rowsPerPage={rowsPerPage}
          page={page}
          onChangePage={handleChangePage}
          onChangeRowsPerPage={handleChangeRowsPerPage}
        />
      </Paper>
      <FormControlLabel
        control={<Switch checked={dense} onChange={handleChangeDense} />}
        label="Dense padding"
      />
    </div>
  );
}
