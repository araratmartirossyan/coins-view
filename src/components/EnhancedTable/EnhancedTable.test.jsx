import React from 'react'
import renderer from 'react-test-renderer'

import EnhancedTable from './EnhancedTable'

it('renders correctly EnhancedTable', () => {
  const input = renderer.create(<EnhancedTable />).toJSON()
  expect(input).toMatchSnapshot()
})
