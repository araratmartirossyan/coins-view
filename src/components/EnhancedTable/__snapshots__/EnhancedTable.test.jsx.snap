// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly EnhancedTable 1`] = `
<div
  className="enhanced-table"
>
  <div
    className="MuiPaper-root enhanced-table__paper MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular tool-bar MuiToolbar-gutters"
    >
      <div
        className="MuiTypography-root tool-bar__title MuiTypography-h6"
        id="tableTitle"
      >
        Информация
      </div>
      <div
        className="MuiBox-root MuiBox-root-64 buttons-controls"
      >
        <div
          className="MuiBox-root MuiBox-root-65 buttons-controls__row"
        >
          <div
            className="MuiBox-root MuiBox-root-66 filters-controls"
          >
            <button
              aria-label="favorites"
              className="MuiButtonBase-root MuiIconButton-root filters-controls__favorite MuiIconButton-colorSecondary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15l-5-2.18L7 18V5h10v13z"
                  />
                </svg>
              </span>
            </button>
            <div
              className="MuiButtonGroup-root filters-controls__buttons-group MuiButtonGroup-disableElevation MuiButtonGroup-contained"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  BNB
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  BTC
                </span>
              </button>
            </div>
            <div
              className="MuiBox-root MuiBox-root-141 filters-controls__selects-group"
            >
              <div
                className="MuiFormControl-root filters-controls__select"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  id="alts-select-label"
                >
                  ALTS
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="alts-select-label alts-select"
                    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="alts-select"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </div>
                  <input
                    onAnimationStart={[Function]}
                    type="hidden"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="MuiFormControl-root filters-controls__select"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                  data-shrink={false}
                  id="usds-select-label"
                >
                  USDS
                </label>
                <div
                  className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                  onClick={[Function]}
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="usds-select-label usds-select"
                    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                    id="usds-select"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </div>
                  <input
                    onAnimationStart={[Function]}
                    type="hidden"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSelect-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-212 buttons-controls__row"
        >
          <div>
            <div
              className="MuiFormControl-root MuiTextField-root"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink={false}
                htmlFor="outlined-basic"
                id="outlined-basic-label"
              >
                Search
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  disabled={false}
                  id="outlined-basic"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-229 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="PrivateNotchedOutline-legendLabelled-231"
                  >
                    <span>
                      Search
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            aria-label="position"
            className="MuiFormGroup-root RadioButtonGroup buttons-controls__radio-group MuiFormGroup-row"
            defaultValue="change"
            role="radiogroup"
          >
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-246 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-247 Mui-checked MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={true}
                    className="PrivateSwitchBase-input-249"
                    name="position"
                    onChange={[Function]}
                    type="radio"
                    value="change"
                  />
                  <div
                    className="PrivateRadioButtonIcon-root-250 PrivateRadioButtonIcon-checked-252"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-251"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Change
              </span>
            </label>
            <label
              className="MuiFormControlLabel-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-246 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-249"
                    name="position"
                    onChange={[Function]}
                    type="radio"
                    value="volume"
                  />
                  <div
                    className="PrivateRadioButtonIcon-root-250"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                      />
                    </svg>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root PrivateRadioButtonIcon-layer-251"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                      />
                    </svg>
                  </div>
                </span>
              </span>
              <span
                className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Volume
              </span>
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiTableContainer-root enhanced-table__container"
    >
      <table
        aria-label="enhanced table"
        aria-labelledby="tableTitle"
        className="MuiTable-root enhanced-table__table"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            >
              <span
                aria-disabled={true}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-246 MuiCheckbox-root MuiCheckbox-colorSecondary PrivateSwitchBase-disabled-248 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    className="PrivateSwitchBase-input-249"
                    data-indeterminate={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-paddingNone"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Pair
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Last Price
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiTableSortLabel-root"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                Change
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
            style={
              Object {
                "height": 265,
              }
            }
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              colSpan={6}
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="MuiTablePagination-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          className="MuiTablePagination-spacer"
        />
        <p
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          Rows per page:
        </p>
        <div
          className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            aria-label="Rows per page:"
            className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </div>
          <input
            onAnimationStart={[Function]}
            type="hidden"
            value={5}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          0-0 of 0
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Previous page"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Next page"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <label
    className="MuiFormControlLabel-root"
  >
    <span
      className="MuiSwitch-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-246 MuiSwitch-switchBase MuiSwitch-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input-249 MuiSwitch-input"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Dense padding
    </span>
  </label>
</div>
`;
